src/log.ts

```diff
@@ -885,61 +867,50 @@ class Log implements LogSeverityFunctions {
     opts?: WriteOptions | ApiResponseCallback
   ): Promise<ApiResponse> {
     const options = opts ? (opts as WriteOptions) : {};
-    // eslint-disable-next-line @typescript-eslint/no-this-alias
-    const self = this;
-    // Autodetect monitored resource if not specified by user in WriteOptions.
+    let decoratedEntries: EntryJson[];
+    // Extract projectId & resource from Logging - inject & memoize if not.
+    await this.logging.setProjectId();
+    this.formattedName_ = formatLogName(this.logging.projectId, this.name);
+    const resource = await this.getOrSetResource(options);
+    // Extract & format additional context from individual entries.
+    try {
+      decoratedEntries = this.decorateEntries(arrify(entry) as Entry[]);
+    } catch (err) {
+      // Ignore errors (the API will speak up if it has an issue).
+    }
+    this.truncateEntries(decoratedEntries!);
+    // Clobber `labels` and `resource` fields with WriteOptions from the user.
+    const reqOpts = extend(
+      {
+        logName: this.formattedName_,
+        entries: decoratedEntries!,
+        resource,
+      },
+      options
+    );
+    delete reqOpts.gaxOptions;
+    return this.logging.loggingService.writeLogEntries(
+      reqOpts,
+      options.gaxOptions
+    );
+  }
+
+  /**
+   * detectResource looks for GCP service context first at the user declaration
```

Comment:
should `detectResource` be `getOrSetResource`?

Reply:
good catch thanks